OpenSim/Examples/DataTable/example.cpp

```diff
@@ -0,0 +1,357 @@
+  dt_vec6_copy = dt_vec6;
+
+  // Add/bind an entire DataTable by col.
+  dt_real.cbindDataTable(dt_real_copy);
+  dt_vec3.cbindDataTable(dt_vec3_copy);
+  dt_vec6.cbindDataTable(dt_vec6_copy);
+
+  // Bind two existing DataTable(s) and produce a new one instead of appending
+  // one to another.
+  {
+    // Construct DataTable of a given size with a default value.
+    decltype(dt_real) dt1_real{3, 4, SimTK::Real{10}};
+    decltype(dt_vec3) dt1_vec3{3, 4, SimTK::Vec3{10, 20, 30}};
+    decltype(dt_vec6) dt1_vec6{3, 4, SimTK::Vec6{10, 20, 30, 40, 50, 60}};
+
+    // Construct DataTable of a given size with a default value.
+    decltype(dt_real) dt2_real{6, 4, SimTK::Real{5}};
+    decltype(dt_vec3) dt2_vec3{6, 4, SimTK::Vec3{5, 6, 7}};
+    decltype(dt_vec6) dt2_vec6{6, 4, SimTK::Vec6{5, 6, 7, 8, 9, 0}};
+
+    // Bind by row.
+    auto rbind_dt_real = rbindDataTables(dt1_real, dt2_real);
+    auto rbind_dt_vec3 = rbindDataTables(dt1_vec3, dt2_vec3);
+    auto rbind_dt_vec6 = rbindDataTables(dt1_vec6, dt2_vec6);
+
+    // Bind by col. The following using same table for both arguments.
+    auto cbind_dt_real = cbindDataTables(dt1_real, dt1_real);
+    auto cbind_dt_vec3 = cbindDataTables(dt1_vec3, dt1_vec3);
+    auto cbind_dt_vec6 = cbindDataTables(dt1_vec6, dt1_vec6);
+  }
+
+  // The columns of a DataTable can be labeled. Not all columns need to be
+  // labeled. 
+  dt_real.insertColLabel(0, "col-zero");
+  dt_vec3.insertColLabel(0, "col-zero");
+  dt_vec6.insertColLabel(0, "col-zero");
+  dt_real.insertColLabel(2, "col-two");
+  dt_vec3.insertColLabel(2, "col-two");
+  dt_vec6.insertColLabel(2, "col-two");
+  dt_real.insertColLabel(4, "col-four");
+  dt_vec3.insertColLabel(4, "col-four");
+  dt_vec6.insertColLabel(4, "col-four");
+
+  // Retrieve number of rows and cols in the DataTable.
+  assert(dt_real.getNumRows() == 14 && dt_real.getNumCols() == 16);
+  assert(dt_vec3.getNumRows() == 14 && dt_vec3.getNumCols() == 16);
+  assert(dt_vec6.getNumRows() == 14 && dt_vec6.getNumCols() == 16);
+
+  // Get a row/col of by its index. Result of get<> functions is not writable.
+  // See SimTK::RowVectorview_ and SimTK::VectorView for details on how to use 
+  // the result. For example, elements of the row/col can be read using 
+  // 'operator[]'.
+  {
+    const auto row2_real = dt_real.getRow(2);
+    const auto row2_vec3 = dt_vec3.getRow(2);
+    const auto row2_vec6 = dt_vec6.getRow(2);
+
+    const auto col2_real = dt_real.getCol(2);
+    const auto col2_vec3 = dt_vec3.getCol(2);
+    const auto col2_vec6 = dt_vec6.getCol(2);
+
+    // Columns can also be retrieved using their labels, if they have been
+    // given one.
+    const auto col2_real_copy = dt_real.getCol("col-two");
+    const auto col2_vec3_copy = dt_vec3.getCol("col-two");
+    const auto col2_vec6_copy = dt_vec6.getCol("col-two");
+    for(int i = 0; i < col2_real.size(); ++i) {
+      assert(std::abs(col2_real[i] - col2_real_copy[i]) < EPSILON);
```

Comment:
This should be an equality test. A copy should preserve every bit. If you are doing this to avoid warnings, you should disable the warning -- the compiler is wrong here.